extensions/pass/src/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { List, getPreferenceValues, Toast, showToast } from "@raycast/api";
+import { useState, useEffect, useCallback } from "react";
+import * as recursive from "recursive-readdir";
+
+import BadPasswordPath from "./errors_handling/bad_password_path";
+import GetPasswordDetails, { passwords_path_structure } from "./passwordDetails";
+import { Preferences } from "./utils";
+
+function ParsePassFileName(pass_file_path: string) {
+  const preferences = getPreferenceValues<Preferences>();
+  const password_store_name = preferences.passwords_store_path;
+  const pass_id = pass_file_path.lastIndexOf(password_store_name);
+  let pass_path = pass_file_path.slice(pass_id);
+  pass_path = pass_path.slice(password_store_name.length).slice(0, -4);
+  return pass_path;
+}
+
+function LoadPassFilesList(PATH_TO_STORE: string) {
+  const [files, setFiles] = useState<passwords_path_structure[]>([{ pass_file_name: "", pass_file_path: "" }]);
+  const [error_loading, setError] = useState<boolean>(false);
```

Comment:
Note that you can use `useCachedPromise` or `usePromise` from `@raycast/utils` if you don't want to bother with errors and loading stuff. This should simplify the code.